### ISSUE TITLE
ACME "autoconfig" Verification failed 

### ISSUE THREAD

**Issue by Ashby666** [mailcow/mailcow-dockerized#3227]
Hi,

since the last update I have trouble getting a certificate from LetsEncrypt.
Log says: "ValueError: Challenge did not pass for".

Full log:

> Sat Dec 21 11:48:37 CET 2019 - Waiting for Docker API...OK
Sat Dec 21 11:48:37 CET 2019 - Waiting for database... Uptime: 1130  Threads: 26  Questions: 1502  Slow queries: 0  Opens: 49  Flush tables: 1  Open tables: 43  Queries per second avg: 1.329
OK
Sat Dec 21 11:48:37 CET 2019 - Waiting for Nginx... OK
Sat Dec 21 11:48:37 CET 2019 - Waiting for domain table... OK
Sat Dec 21 11:48:37 CET 2019 - Initializing, please wait...
Sat Dec 21 11:48:37 CET 2019 - Using existing domain rsa key /var/lib/acme/acme/key.pem
Sat Dec 21 11:48:37 CET 2019 - Using existing Lets Encrypt account key /var/lib/acme/acme/account.pem
Sat Dec 21 11:48:37 CET 2019 - Detecting IP addresses... OK
Sat Dec 21 11:49:19 CET 2019 - Found A record for autodiscover.domain6: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:19 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:21 CET 2019 - No A or AAAA record found for hostname autoconfig.domain6
Sat Dec 21 11:49:24 CET 2019 - Found A record for autodiscover.domain8: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:24 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:25 CET 2019 - Found A record for autoconfig.domain8: 89.31.143.1
(skipping check, returning 0)
Sat Dec 21 11:49:25 CET 2019 - Confirmed A record 89.31.143.1
Sat Dec 21 11:49:25 CET 2019 - Found A record for autodiscover.domain1: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:25 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:25 CET 2019 - Found AAAA record for autoconfig.domain1: ipv6-address - skipping A record check
(skipping check, returning 0)
Sat Dec 21 11:49:25 CET 2019 - Confirmed AAAA record with IP ipv6-address
Sat Dec 21 11:49:26 CET 2019 - Found A record for autodiscover.domain4: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:26 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:27 CET 2019 - No A or AAAA record found for hostname autoconfig.domain4
Sat Dec 21 11:49:27 CET 2019 - Found A record for autodiscover.domain5: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:27 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:30 CET 2019 - No A or AAAA record found for hostname autoconfig.domain5
Sat Dec 21 11:49:30 CET 2019 - Found A record for autodiscover.domain2: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:30 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:32 CET 2019 - Found AAAA record for autoconfig.domain2: ipv6-address - skipping A record check
(skipping check, returning 0)
Sat Dec 21 11:49:32 CET 2019 - Confirmed AAAA record with IP ipv6-address
Sat Dec 21 11:49:33 CET 2019 - Found A record for autodiscover.domain4: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:33 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:34 CET 2019 - No A or AAAA record found for hostname autoconfig.domain4
[root@PJ-Docker03-Mail mailcow-dockerized]# docker logs mailcowdockerized_acme-mailcow_1
Sat Dec 21 11:29:40 CET 2019 - Waiting for Docker API...OK
Sat Dec 21 11:29:40 CET 2019 - Waiting for database... Uptime: 1  Threads: 8  Questions: 2  Slow queries: 0  Opens: 17  Flush tables: 1  Open tables: 11  Queries per second avg: 2.000
OK
Sat Dec 21 11:29:40 CET 2019 - Waiting for Nginx... OK
Sat Dec 21 11:29:42 CET 2019 - Waiting for domain table... OK
Sat Dec 21 11:29:42 CET 2019 - Initializing, please wait...
Sat Dec 21 11:29:42 CET 2019 - Using existing domain rsa key /var/lib/acme/acme/key.pem
Sat Dec 21 11:29:42 CET 2019 - Using existing Lets Encrypt account key /var/lib/acme/acme/account.pem
Sat Dec 21 11:29:42 CET 2019 - Detecting IP addresses... OK
Sat Dec 21 11:30:29 CET 2019 - Found A record for autodiscover.domain6: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:29 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:30 CET 2019 - No A or AAAA record found for hostname autoconfig.domain6
Sat Dec 21 11:30:30 CET 2019 - Found A record for autodiscover.domain8: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:30 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:30 CET 2019 - Found A record for autoconfig.domain8: 89.31.143.1
(skipping check, returning 0)
Sat Dec 21 11:30:30 CET 2019 - Confirmed A record 89.31.143.1
Sat Dec 21 11:30:30 CET 2019 - Found A record for autodiscover.domain1: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:30 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:30 CET 2019 - Found AAAA record for autoconfig.domain1: ipv6-address - skipping A record check
(skipping check, returning 0)
Sat Dec 21 11:30:30 CET 2019 - Confirmed AAAA record with IP ipv6-address
Sat Dec 21 11:30:34 CET 2019 - Found A record for autodiscover.domain4: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:34 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:34 CET 2019 - No A or AAAA record found for hostname autoconfig.domain4
Sat Dec 21 11:30:40 CET 2019 - Found A record for autodiscover.domain5: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:40 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:40 CET 2019 - No A or AAAA record found for hostname autoconfig.domain5
Sat Dec 21 11:30:41 CET 2019 - Found A record for autodiscover.domain2: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:41 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:42 CET 2019 - Found AAAA record for autoconfig.domain2: ipv6-address - skipping A record check
(skipping check, returning 0)
Sat Dec 21 11:30:42 CET 2019 - Confirmed AAAA record with IP ipv6-address
Sat Dec 21 11:30:42 CET 2019 - Found A record for autodiscover.domain4: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:42 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:45 CET 2019 - No A or AAAA record found for hostname autoconfig.domain4
Sat Dec 21 11:30:45 CET 2019 - No A or AAAA record found for hostname autodiscover.domain7
Sat Dec 21 11:30:46 CET 2019 - No A or AAAA record found for hostname autoconfig.domain7
Sat Dec 21 11:30:46 CET 2019 - Found A record for autodiscover.domain3: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:46 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:48 CET 2019 - No A or AAAA record found for hostname autoconfig.domain3
Sat Dec 21 11:30:48 CET 2019 - Found A record for mail.domain1: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:30:48 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:30:48 CET 2019 - Certificate /var/lib/acme/mail.domain1/cert.pem missing or changed domains 'mail.domain1 autoconfig.domain8 autoconfig.domain1 autoconfig.domain2 autodiscover.domain6 autodiscover.domain8 autodiscover.domain1 autodiscover.domain4 autodiscover.domain5 autodiscover.domain2 autodiscover.domain4 autodiscover.domain3' - start obtaining
Sat Dec 21 11:30:48 CET 2019 - Creating backups in /var/lib/acme/backups/mail.domain1/2019-12-21_11_30_48 ...
Parsing account key...
Parsing CSR...
Found domains: autodiscover.domain4, autodiscover.domain3, autodiscover.domain2, autoconfig.domain1, autoconfig.domain8, autodiscover.domain1, autodiscover.domain5, autodiscover.domain8, autodiscover.domain4, autodiscover.domain6, autoconfig.domain2, mail.domain1
Getting directory...
Directory found!
Registering account...
Already registered!
Creating new order...
Order created!
Verifying autoconfig.domain8...
Traceback (most recent call last):
  File "/usr/bin/acme-tiny", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3.7/site-packages/acme_tiny.py", line 194, in main
    signed_crt = get_crt(args.account_key, args.csr, args.acme_dir, log=LOGGER, CA=args.ca, disable_check=args.disable_check, directory_url=args.directory_url, contact=args.contact)
  File "/usr/lib/python3.7/site-packages/acme_tiny.py", line 149, in get_crt
    raise ValueError("Challenge did not pass for {0}: {1}".format(domain, authorization))
ValueError: Challenge did not pass for autoconfig.domain8: {'identifier': {'type': 'dns', 'value': 'autoconfig.domain8'}, 'status': 'invalid', 'expires': '2019-12-28T10:11:50Z', 'challenges': [{'type': 'http-01', 'status': 'invalid', 'error': {'type': 'urn:ietf:params:acme:error:unauthorized', 'detail': 'Invalid response from http://autoconfig.domain8/.well-known/acme-challenge/wPue1gVT0HpQrkLFlx5Df6dkowUIGtoGZAQaMBmauv8 [89.31.143.1]: "<!DOCTYPE html> <html lang=\\"de\\">  <head> <meta name=\\"description\\" content=\\"Hier entsteht domain8\\"> <meta http-equiv=\\"Conten"', 'status': 403}, 'url': 'https://acme-v02.api.letsencrypt.org/acme/chall-v3/1848173120/NsHZlg', 'token': 'wPue1gVT0HpQrkLFlx5Df6dkowUIGtoGZAQaMBmauv8', 'validationRecord': [{'url': 'http://autoconfig.domain8/.well-known/acme-challenge/wPue1gVT0HpQrkLFlx5Df6dkowUIGtoGZAQaMBmauv8', 'hostname': 'autoconfig.domain8', 'port': '80', 'addressesResolved': ['89.31.143.1'], 'addressUsed': '89.31.143.1'}]}, {'type': 'dns-01', 'status': 'invalid', 'url': 'https://acme-v02.api.letsencrypt.org/acme/chall-v3/1848173120/oORR8g', 'token': 'wPue1gVT0HpQrkLFlx5Df6dkowUIGtoGZAQaMBmauv8'}, {'type': 'tls-alpn-01', 'status': 'invalid', 'url': 'https://acme-v02.api.letsencrypt.org/acme/chall-v3/1848173120/CiFcDQ', 'token': 'wPue1gVT0HpQrkLFlx5Df6dkowUIGtoGZAQaMBmauv8'}]}
Sat Dec 21 11:30:59 CET 2019 - Failed to obtain certificate /var/lib/acme/mail.domain1/cert.pem for domains 'mail.domain1 autoconfig.domain8 autoconfig.domain1 autoconfig.domain2 autodiscover.domain6 autodiscover.domain8 autodiscover.domain1 autodiscover.domain4 autodiscover.domain5 autodiscover.domain2 autodiscover.domain4 autodiscover.domain3'
OK
Sat Dec 21 11:30:59 CET 2019 - Some errors occurred, retrying in 30 minutes...
OK
Sat Dec 21 11:48:37 CET 2019 - Waiting for Docker API...OK
Sat Dec 21 11:48:37 CET 2019 - Waiting for database... Uptime: 1130  Threads: 26  Questions: 1502  Slow queries: 0  Opens: 49  Flush tables: 1  Open tables: 43  Queries per second avg: 1.329
OK
Sat Dec 21 11:48:37 CET 2019 - Waiting for Nginx... OK
Sat Dec 21 11:48:37 CET 2019 - Waiting for domain table... OK
Sat Dec 21 11:48:37 CET 2019 - Initializing, please wait...
Sat Dec 21 11:48:37 CET 2019 - Using existing domain rsa key /var/lib/acme/acme/key.pem
Sat Dec 21 11:48:37 CET 2019 - Using existing Lets Encrypt account key /var/lib/acme/acme/account.pem
Sat Dec 21 11:48:37 CET 2019 - Detecting IP addresses... OK
Sat Dec 21 11:49:19 CET 2019 - Found A record for autodiscover.domain6: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:19 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:21 CET 2019 - No A or AAAA record found for hostname autoconfig.domain6
Sat Dec 21 11:49:24 CET 2019 - Found A record for autodiscover.domain8: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:24 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:25 CET 2019 - Found A record for autoconfig.domain8: 89.31.143.1
(skipping check, returning 0)
Sat Dec 21 11:49:25 CET 2019 - Confirmed A record 89.31.143.1
Sat Dec 21 11:49:25 CET 2019 - Found A record for autodiscover.domain1: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:25 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:25 CET 2019 - Found AAAA record for autoconfig.domain1: ipv6-address - skipping A record check
(skipping check, returning 0)
Sat Dec 21 11:49:25 CET 2019 - Confirmed AAAA record with IP ipv6-address
Sat Dec 21 11:49:26 CET 2019 - Found A record for autodiscover.domain4: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:26 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:27 CET 2019 - No A or AAAA record found for hostname autoconfig.domain4
Sat Dec 21 11:49:27 CET 2019 - Found A record for autodiscover.domain5: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:27 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:30 CET 2019 - No A or AAAA record found for hostname autoconfig.domain5
Sat Dec 21 11:49:30 CET 2019 - Found A record for autodiscover.domain2: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:30 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:32 CET 2019 - Found AAAA record for autoconfig.domain2: ipv6-address - skipping A record check
(skipping check, returning 0)
Sat Dec 21 11:49:32 CET 2019 - Confirmed AAAA record with IP ipv6-address
Sat Dec 21 11:49:33 CET 2019 - Found A record for autodiscover.domain4: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:33 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:34 CET 2019 - No A or AAAA record found for hostname autoconfig.domain4
Sat Dec 21 11:49:39 CET 2019 - No A or AAAA record found for hostname autodiscover.domain7
Sat Dec 21 11:49:40 CET 2019 - No A or AAAA record found for hostname autoconfig.domain7
Sat Dec 21 11:49:40 CET 2019 - Found A record for autodiscover.domain3: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:40 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:44 CET 2019 - No A or AAAA record found for hostname autoconfig.domain3
Sat Dec 21 11:49:44 CET 2019 - Found A record for mail.domain1: 62.112.xx.xxx
(skipping check, returning 0)
Sat Dec 21 11:49:44 CET 2019 - Confirmed A record 62.112.xx.xxx
Sat Dec 21 11:49:44 CET 2019 - Certificate /var/lib/acme/mail.domain1/cert.pem missing or changed domains 'mail.domain1 autoconfig.domain8 autoconfig.domain1 autoconfig.domain2 autodiscover.domain6 autodiscover.domain8 autodiscover.domain1 autodiscover.domain4 autodiscover.domain5 autodiscover.domain2 autodiscover.domain4 autodiscover.domain3' - start obtaining
Sat Dec 21 11:49:44 CET 2019 - Creating backups in /var/lib/acme/backups/mail.domain1/2019-12-21_11_49_44 ...
Parsing account key...
Parsing CSR...
Found domains: autodiscover.domain8, autodiscover.domain1, autoconfig.domain2, autoconfig.domain8, autodiscover.domain5, autoconfig.domain1, autodiscover.domain2, autodiscover.domain6, autodiscover.domain3, mail.domain1, autodiscover.domain4, autodiscover.domain4
Getting directory...
Directory found!
Registering account...
Already registered!
Creating new order...
Order created!
Verifying autoconfig.domain1...
Traceback (most recent call last):
  File "/usr/bin/acme-tiny", line 8, in <module>
    sys.exit(main())
  File "/usr/lib/python3.7/site-packages/acme_tiny.py", line 194, in main
    signed_crt = get_crt(args.account_key, args.csr, args.acme_dir, log=LOGGER, CA=args.ca, disable_check=args.disable_check, directory_url=args.directory_url, contact=args.contact)
  File "/usr/lib/python3.7/site-packages/acme_tiny.py", line 149, in get_crt
    raise ValueError("Challenge did not pass for {0}: {1}".format(domain, authorization))
ValueError: Challenge did not pass for autoconfig.domain1: {'identifier': {'type': 'dns', 'value': 'autoconfig.domain1'}, 'status': 'invalid', 'expires': '2019-12-28T10:30:55Z', 'challenges': [{'type': 'http-01', 'status': 'invalid', 'error': {'type': 'urn:ietf:params:acme:error:unauthorized', 'detail': 'Invalid response from http://autoconfig.domain1/.well-known/acme-challenge/RT-O52LhbxNxl5y5efwaOY7-WSr3tiGxXECrD--O8hg [ipv6-address]: 404', 'status': 403}, 'url': 'https://acme-v02.api.letsencrypt.org/acme/chall-v3/1848392219/RM2aAQ', 'token': 'RT-O52LhbxNxl5y5efwaOY7-WSr3tiGxXECrD--O8hg', 'validationRecord': [{'url': 'http://autoconfig.domain1/.well-known/acme-challenge/RT-O52LhbxNxl5y5efwaOY7-WSr3tiGxXECrD--O8hg', 'hostname': 'autoconfig.domain1', 'port': '80', 'addressesResolved': ['81.169.145.141', 'ipv6-address'], 'addressUsed': 'ipv6-address'}]}, {'type': 'dns-01', 'status': 'invalid', 'url': 'https://acme-v02.api.letsencrypt.org/acme/chall-v3/1848392219/z42mKg', 'token': 'RT-O52LhbxNxl5y5efwaOY7-WSr3tiGxXECrD--O8hg'}, {'type': 'tls-alpn-01', 'status': 'invalid', 'url': 'https://acme-v02.api.letsencrypt.org/acme/chall-v3/1848392219/K4poJw', 'token': 'RT-O52LhbxNxl5y5efwaOY7-WSr3tiGxXECrD--O8hg'}]}
Sat Dec 21 11:49:59 CET 2019 - Failed to obtain certificate /var/lib/acme/mail.domain1/cert.pem for domains 'mail.domain1 autoconfig.domain8 autoconfig.domain1 autoconfig.domain2 autodiscover.domain6 autodiscover.domain8 autodiscover.domain1 autodiscover.domain4 autodiscover.domain5 autodiscover.domain2 autodiscover.domain4 autodiscover.domain3'
OK
Sat Dec 21 11:49:59 CET 2019 - Some errors occurred, retrying in 30 minutes...

In my investigation I found this post #2736 and this [answer ](https://github.com/mailcow/mailcow-dockerized/issues/2736#issuecomment-506942739)from andryyy. Unfortunately it does not solve the problem, because the lines from the answer post are not present.

Here the code from _data/conf/nginx/site.conf_ without any changes.

> [root@PJ-Docker03-Mail mailcow-dockerized]# cat data/conf/nginx/site.conf | head -n 100
server_tokens off;
proxy_cache_path /tmp levels=1:2 keys_zone=sogo:10m inactive=24h  max_size=1g;
server_names_hash_bucket_size 64;
map $http_x_forwarded_proto $client_req_scheme {
     default $scheme;
     https https;
}
include /etc/nginx/conf.d/sites.active;

Do you guys have an idea, how I can solve this problem?

Thank you!

**Comment by andryyy** [comment]
You skip all checks for ACME validation in mailcow. It was told to not care to verify the config. You need to figure out which domain does not validate and reconfigure the webserver/DNS/firewall/whatever to complete the ACME challenge successfully. 

 Check autoconfig.domain8

**Comment by andryyy** [comment]
This is also not related to an update. It is just a broken configuration. The update does not change your DNS etc. :)

**Comment by Ashby666** [comment]
Hi andryy and thank you very much so far :-)

Unfortunately I can not create an DNS-record vor autoconfig.domain8 because it is not allowed. The hosting provider doesn't accept it.  The cow had been working for over half a year without changing the configuration. So where was the change? Probably at the provider, right?

So the only solution is to move the domain? I do not need autoconfig since everybody uses Active Sync.

**Comment by andryyy** [comment]
Why wouldn't they accept it? :)
There seems to be a wildcard active, that catches autoconfig.domain8. You disabled the verifications (SKIP_IP_CHECK or SKIP_HTTP_VERIFICATION are set to y). If you re-enable these checks, mailcow will fail to verify the names and skip them.

**Comment by Ashby666** [comment]
Hi andryyy,

hope you had some nice days off!
I moved the domain over christmas from "Strato" to another hoster where I was able to set autoconfig.. Everything works now fine again.

Even if only the domain package is booked at Strato, they now reserve autoconfig. and put a CNAME on it. For what I unfortunately do not know ... if you have booked a mailbox package that certainly makes sense. But otherwise... ?

However. My problem is solved. Thank you very much!
